core/src/services/ghaa/core.rs

```diff
@@ -0,0 +1,185 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fmt::{Debug, Formatter};
+
+use crate::raw::HttpClient;
+use http::header::ACCEPT;
+use http::header::AUTHORIZATION;
+use http::header::USER_AGENT;
+use http::Request;
+use http::Response;
+
+use crate::raw::*;
+use crate::services::GhaaConfig;
+use crate::*;
+
+/// VERSION is the compiled version of OpenDAL.
+pub const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+const HEADER_GITHUB_API_VERSION: &str = "X-GitHub-Api-Version";
+
+/// The default endpoint suffix for ghaa.
+const DEFAULT_GHAA_ENDPOINT_SUFFIX: &str = "https://api.github.com/repos";
+
+pub struct GhaaCore {
+    pub config: GhaaConfig,
+    pub client: HttpClient,
+}
+
+impl Debug for GhaaCore {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("GhaaCore")
+            .field("owner", &self.config.owner)
+            .field("repo", &self.config.repo)
+            .finish_non_exhaustive()
+    }
+}
+impl GhaaCore {
+    pub async fn download_artifact(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url: String = format!(
+            "{}/{}/{}/actions/artifacts/{}/zip",
```

Comment:
The content must be mismatch if we download as a zip.

Reply:
https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#download-an-artifact

> The :archive_format must be zip.

Reply:
It's acceptable for services to always return `zip` files, but we should unzip them to provide users with readable raw content.

Reply:
Hi xuanwo, I think I need some help about this.
```
let body = resp.into_body();
// body is .zip format, we need unzip it and return the raw content
let mut zip = ZipArchive::new(body).map_err(|err| {
                    Error::new(ErrorKind::Unexpected, &format!("unzip error: {}", err))})?;
Ok((RpRead::new(), ))
```
What should I return here, I got a lot of files from the `.zip` file

Reply:
Certainly. This is indeed a serious issue as GitHub artifacts may contain multiple files that are incompatible with opendal's API.